cinema/views.py

```diff
@@ -0,0 +1,58 @@
+from django.http import JsonResponse, HttpResponse
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.parsers import JSONParser
+from rest_framework.response import Response
+
+from cinema.models import Movie
+from cinema.serializers import MovieSerializer
+
+
+@api_view(["GET", "POST"])
+def movie_list(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    if request.method == "POST":
+        serializer = MovieSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(
+                serializer.data,
+                status=status.HTTP_201_CREATED
+            )
+        return Response(
+            serializer.errors,
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk):
+    """
+    Retrieve, update or delete a code movie.
+    """
+
+    movies = get_object_or_404(Movie, pk=pk)
+
+    if request.method == "GET":
+        serializer = MovieSerializer(movies)
+        return Response(serializer.data)
+
+    elif request.method == "PUT":
+        data = JSONParser().parse(request)
+        serializer = MovieSerializer(movies, data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(serializer.data)
+        return JsonResponse(
+            serializer.errors,
+            status=status.HTTP_400_BAD_REQUEST
+        )
```

Comment:
use is_valid(raise_exception=True) to do all checks automatically and raise handled errors if it's needed